File: java/src/main/java/com/cillu/ports/TeamServicePort.java
package com.cillu.ports;

import com.cillu.core.domain.Player;
import com.cillu.core.domain.Team;

import java.util.List;

public interface TeamServicePort {
    Team createNewTeam(Team team) throws Exception;
    List<Team> listTeams() throws Exception;
    Player invitePlayer(Player player) throws Exception;
    List<Player> listTeamPlayers(int teamID) throws Exception;
}

File: ports/team_service_port.go
package ports

import domain "github.com/thecillu/exagonal-architecture/core/domain"

type TeamServicePort interface {
	CreateNewTeam(team domain.Team) (domain.Team, error)
	ListTeams() ([]domain.Team, error)
	InvitePlayer(player domain.Player) (domain.Player, error)
	ListTeamPlayers(team_id int64) ([]domain.Player, error)
}
